internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -278,6 +278,20 @@ func IsSharedTier(instanceSize string) bool {
 	return instanceSize == "M0" || instanceSize == "M2" || instanceSize == "M5"
 }
 
+// GetDiskSizeGBFromReplicationSpec obtains the diskSizeGB value by looking into the electable spec of the first replication spec.
+// Independent storage size scaling is not supported (CLOUDP-201331), meaning all electable/analytics/readOnly configs in all replication specs are the same.
+func GetDiskSizeGBFromReplicationSpec(cluster *admin.ClusterDescription20250101) float64 {
+	specs := cluster.GetReplicationSpecs()
+	if len(specs) < 1 {
+		return 0
+	}
+	configs := specs[0].GetRegionConfigs()
+	if len(configs) < 1 {
+		return 0
+	}
+	return configs[0].ElectableSpecs.GetDiskSizeGB()
```

Comment:
is there always an electable spec, even if not defined in the TF file?

Reply:
Correct, the API always returns `electableSpecs`, `readOnlySpecs`, and `analyticsSpecs` even if not defined in the request. Example of readOnlySpecs response when not defined:
```
"readOnlySpecs": {
    "instanceSize": "M30",
    "diskIOPS": 120,
    "diskSizeGB": 32.0,
    "diskThroughput": 25,
    "nodeCount": 0
}
```